data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/config/BaseDb2Config.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.americanexpress.synapse.data.db2.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.core.env.Environment;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import javax.sql.DataSource;
+import java.util.Properties;
+
+/**
+ * {@code BaseDb2Config} contains base level configurations for db2 required for all usage of this data module.
+ * @author tisla4
+ */
+@Configuration
+@EnableTransactionManagement
+public abstract class BaseDb2Config {
+
+    /**
+     * Contains environment variables needed for connection.
+     */
+    @Autowired
+    protected Environment environment;
```

Comment:
is it possible to do constructor autowiring?

private final Environment environment;

public BaseDb2Condif(Environment environment) { this.environment = environment; }

Reply:
done